lucille-core/src/main/java/com/kmwllc/lucille/core/PublisherImpl.java

```diff
@@ -93,6 +94,7 @@ public PublisherImpl(Config config, PublisherMessenger messenger, String runId,
 
   @Override
   public void publish(Document document) throws Exception {
+    MDC.put("docId", document.getId());
```

Comment:
in local mode, the publisher will be running in one thread, some number of workers will be running in their own threads, and an indexer will be running in yet another thread. I don't think setting the docId in the MDC here will propagate the change to other threads where docId is needed, but even if it did, the propogated docId would be "right" for those other threads because everything is happening concurrently -- one doc3 might be published at the same time worker 1 is processing doc2 and the indexer is indexing doc1...